### PR TITLE
Add API for specifying the system overlay style.

### PR DESCRIPTION
**Pull request by tvolkert**
Fixes 3544

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/app.dart [hunk 1]
```diff
@@ -65,6 +67,13 @@ class GalleryAppState extends State<GalleryApp> {
   bool _useLightTheme = true;
   bool _showPerformanceOverlay = false;
 
+  void _onThemeChanged(bool useLightTheme) {
+    SystemChrome.setSystemUIOverlayStyle(useLightTheme
+        ? mojom.SystemUiOverlayStyle.dark
+        : mojom.SystemUiOverlayStyle.light);
+    setState(() { _useLightTheme = useLightTheme; });
+  }
```
Comment (abarth) [inline examples/flutter_gallery/lib/gallery/app.dart]: We should make this part of the `Title` widget.

Comment (tvolkert) [inline examples/flutter_gallery/lib/gallery/app.dart]: Done

File: packages/flutter/lib/src/widgets/title.dart [hunk 2]
```diff
@@ -28,10 +33,22 @@ class Title extends StatelessWidget {
...
+  /// Updates the system chrome settings based on this title's metadata.
+  void _updateSystemChrome() {
+    // TODO(tvolkert): This may result in a decent amount of unnecessary
+    // overhead in the embedder (on every build). Consider making Title
+    // a StatefulWidget that only calls into the SystemChrome service when
+    // it sees that a value has changed.
+    SystemChrome.setSystemUIOverlayStyle(brightness == Brightness.dark
+        ? mojom.SystemUiOverlayStyle.light
+        : mojom.SystemUiOverlayStyle.dark);
+  }
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/title.dart]: Yeah, `updateTaskDescription` has logic to filter out redundant writes in this way.

**Comment by abarth** [comment]
LGTM

**Pull request merged**